cpp/src/arrow/compute/exec/plan_test.cc

```diff
@@ -301,11 +301,11 @@ TEST(ExecPlan, ToString) {
                     {"sink", SinkNodeOptions{&sink_gen}},
                 })
                 .AddToPlan(plan.get()));
-  EXPECT_EQ(plan->sources()[0]->ToString(), R"(:SourceNode{outputs=[:SinkNode]})");
-  EXPECT_EQ(plan->sinks()[0]->ToString(), R"(:SinkNode{inputs=[collected=:SourceNode]})");
+  EXPECT_EQ(plan->sources()[0]->ToString(), R"(:SourceNode{})");
+  EXPECT_EQ(plan->sinks()[0]->ToString(), R"(:SinkNode{})");
   EXPECT_EQ(plan->ToString(), R"(ExecPlan with 2 nodes:
-:SourceNode{outputs=[:SinkNode]}
-:SinkNode{inputs=[collected=:SourceNode]}
+:SinkNode{}
+  :SourceNode{}
```

Comment:
I'm curious, why are there colons at the start of the nodes?

Reply:
The format is label: kind_node, so when a node has no label, only: kind_node remains. We could prevent the colons from appearing when there is no label.

Reply:
The format is label: kind_node, so when a node has no label, only: kind_node remains. We could prevent the colons from appearing when there is no label.

Reply:
I would not worry about it.

`ExecPlan::AddNode` will attach a label (auto-incrementing id counter) if none is set so there should not be any valid cases where there is no label.